Orca Integration (#323)
solana-developers/solana-cookbook: docs/integrations/orca.md

---
title: Orca
head:
  - - meta
    - name: title
      content: Solana Cookbook | Integrating Orca in your App
  - - meta
    - name: og:title
      content: Solana Cookbook | Integrating Orca in your App
  - - meta
    - name: description
      content: Orca is the easiest place to exchange cryptocurrency on the Solana blockchain. On Orca, you can exchange tokens more cheaply, quickly, and confidently.
  - - meta
    - name: og:description
      content: Orca is the easiest place to exchange cryptocurrency on the Solana blockchain. On Orca, you can exchange tokens more cheaply, quickly, and confidently.
  - - meta
    - name: og:image
      content: https://solanacookbook.com/cookbook-sharing-card.png
  - - meta
    - name: og:image:alt
      content: Solana splash card
  - - meta
    - name: twitter:card
      content: summary
  - - meta
    - name: twitter:site
      content: "@solanacookbook"
  - - meta
    - name: twitter:image
      content: "https://solanacookbook.com/cookbook-sharing-card.png"
  - - meta
    - name: robots
      content: index,follow,noodp
  - - meta
    - name: googlebot
      content: index,follow
---

# Orca

> Orca is a Dex, which helps to exchange cryptocurrencies on the Solana Blockchain as quickly, cheaply and confidently as possible.

Reviewer: ```suggestion
Orca is a way to exchange cryptocurrency on the Solana blockchain.
```

Reviewer: This is directly from their site, and avoids words that would cause bias.

You can start interacting with orca program by using Orca Javascript SDK and the @solana/web3.js library.

## Installing Dependencies

> 

Reviewer: Need some information about this section, just a little blurb. Also, why `decimal.js`?

<CodeGroup>
  <CodeGroupItem title="YARN" active>

```bash
yarn add @orca-so/sdk @solana/web3.js decimal.js
```

  </CodeGroupItem>

  <CodeGroupItem title="NPM">

```bash
npm install @orca-so/sdk @solana/web3.js decimal.js
```

  </CodeGroupItem>
</CodeGroup>

## Initialization Orca Object

> In this Code we are initializing the orca object with a connection pointing to mainnet.

Reviewer: Better to explain the Orca object and what it is used for.

<SolanaCodeGroup>
  <SolanaCodeGroupItem title="TS" active>

  <template v-slot:default>

@[code](@/code/orca/init/main.en.ts)

  </template>

  <template v-slot:preview>

@[code](@/code/orca/init/main.preview.en.ts)

  </template>

  </SolanaCodeGroupItem>

</SolanaCodeGroup>

## Performing Swap on Orca

> We use the orca Object to get a pool object and all the supported pools by orca are under the OrcaPoolConfig object. We use the pool object to swap tokens for that pool.

Reviewer: A user will come into this not know what a pool is. May be a good place to explain

<SolanaCodeGroup>
  <SolanaCodeGroupItem title="TS" active>

  <template v-slot:default>

@[code](@/code/orca/swap/main.en.ts)

  </template>

  <template v-slot:preview>

@[code](@/code/orca/swap/main.preview.en.ts)

  </template>

  </SolanaCodeGroupItem>

</SolanaCodeGroup>

## Deposit X and Y tokens to get X-Y LP tokens

> Here we are using the same pool object defined before to deposit token X and Y to the pool and get X-Y LP tokens after depositing.

Reviewer: Same here with LP. Explain without acronyms what it means to deposit into a liquidity pool. Orca should have plenty of this information on their `learn` site.

<SolanaCodeGroup>
  <SolanaCodeGroupItem title="TS" active>

  <template v-slot:default>

@[code](@/code/orca/get_LP/main.en.ts)

  </template>

  <template v-slot:preview>

@[code](@/code/orca/get_LP/main.preview.en.ts)

  </template>

  </SolanaCodeGroupItem>

</SolanaCodeGroup>

## Farming of X-Y LP tokens

> In this code the X-Y LP tokens are deposited to a Farm to earn some rewards.

Reviewer: Also explain farming here

<SolanaCodeGroup>
  <SolanaCodeGroupItem title="TS" active>

  <template v-slot:default>

@[code](@/code/orca/farm/main.en.ts)

  </template>

  <template v-slot:preview>

@[code](@/code/orca/farm/main.preview.en.ts)

  </template>

  </SolanaCodeGroupItem>

</SolanaCodeGroup>

## Withdraw X-Y LP tokens from a farm

> Here we are using the pool object and the owner to withdraw the LP token's deposited in the Farm.

Reviewer: What does it mean to withdraw liquidity from a farm?

<SolanaCodeGroup>
  <SolanaCodeGroupItem title="TS" active>

  <template v-slot:default>

@[code](@/code/orca/farm_withdraw/main.en.ts)

  </template>

  <template v-slot:preview>

@[code](@/code/orca/farm_withdraw/main.preview.en.ts)

  </template>

  </SolanaCodeGroupItem>

</SolanaCodeGroup>

## Withdraw X and Y tokens, from X-Y LP tokens

> Here we are using the pool object and the owner to withdraw X and Y token's deposited in the pool.

Reviewer: What does it mean to unwrap the liquidity pool tokens into their separate tokens?
